math/factorial/factorial_test.go

```diff
@@ -1,81 +1,75 @@
 // factorial_test.go
 // description: Test for calculating factorial
-// author(s) [red_byte](https://github.com/i-redbyte)
 // see factorial.go
 
 package factorial
 
 import "testing"
+import "fmt"
 
-func getTests() []struct {
-	name   string
-	n      int
-	result int
-} {
-	var tests = []struct {
-		name   string
-		n      int
-		result int
-	}{
-		{"5!", 5, 120},
-		{"3!", 3, 6},
-		{"6!", 6, 720},
-		{"12!", 12, 479001600},
-		{"1!", 1, 1},
-	}
-	return tests
-}
+type factorialFun func(int) (int, error)
 
-func TestBruteForceFactorial(t *testing.T) {
-	tests := getTests()
-	for _, tv := range tests {
-		t.Run(tv.name, func(t *testing.T) {
-			result := Iterative(tv.n)
-			if result != tv.result {
-				t.Errorf("Wrong result! Expected:%d, returned:%d ", tv.result, result)
-			}
-		})
+func implementations() map[string]factorialFun {
```

Comment:
I think you can define it as a variable inside this test package if you change it to `factorial_test` (this is also fine and one more function call is not gonna affect the performance). I generally ask a question when it comes to these things, is this return value always going to remain static? If so, using un-exported `var` makes sense. If it changes on each function, keep it inside a function because we don't want side-effects.

Reply:
I like to have _immutable_ constants. In any case, have a look at 66e653933c45e95dce442386d043183c9fa5f7b1.